fmf/cli.py

```diff
@@ -124,7 +124,7 @@ def main(cmdline=None):
             show = node.show(
                 options.brief, options.formatting, options.values)
             if show is not None:
-                print(show, end="")
+                print(show.encode('utf-8'), end="")
```

Comment:
Guys, this is broken.  Take a look at what happens on Python 3 (tested on 7486060d76b3564d3e83cf8ed5e9e7b3df6c62d7):

```
$ fmf
b'\x1b[0;31mfmf/examples/deep/one/two/three\x1b[1;m\n\x1b[0;33mkey\x1b[1;m: value\n'b'\x1b[0;31mfmf/examples/merge/parent/child\x1b[1;m\n\x1b[0;33mdescription\x1b[1;m: General descriptionSpecific description\n\x1b[0;33mtags\x1b[1;m: Tier1 and Tier2\n\x1b[0;33mtime\x1b[1;m: 15\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/bind-address\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/continue\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/get-file\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/download\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/no-clobber\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/download\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/output-document\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/download\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/progress\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/quota\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/server-response\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/spider\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/timestamping\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/requirements/tries\x1b[1;m\n\x1b[0;33mpriority\x1b[1;m: low\n\x1b[0;33mrequirement\x1b[1;m: Various download options working correctly\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/download/test\x1b[1;m\n\x1b[0;33mdescription\x1b[1;m: Check basic download options\n\x1b[0;33mtags\x1b[1;m: Tier1 and TierSecurity\n\x1b[0;33mtest\x1b[1;m: runtest.sh\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n\x1b[0;33mtime\x1b[1;m: 3 min\n'b'\x1b[0;31mfmf/examples/wget/protocols/ftp\x1b[1;m\n\x1b[0;33mdescription\x1b[1;m: Test file download using the ftp protocol.\n\x1b[0;33mdetails\x1b[1;m: This is a very long and detailed description of what the wget/protocols/ftp test actually does including multiple examples and information about proper environment setup.\n\x1b[0;33mtags\x1b[1;m: Tier1\n\x1b[0;33mtest\x1b[1;m: runtest.sh\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n\x1b[0;33mtime\x1b[1;m: 1 min\n'b'\x1b[0;31mfmf/examples/wget/protocols/http\x1b[1;m\n\x1b[0;33mdescription\x1b[1;m: Test file download using the http protocol.\n\x1b[0;33mdetails\x1b[1;m: This is a very long and detailed description of what the wget/protocols/http test actually does including multiple examples and information about proper environment setup.\n\x1b[0;33mtags\x1b[1;m: Tier1\n\x1b[0;33mtest\x1b[1;m: runtest.sh\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n\x1b[0;33mtime\x1b[1;m: 1 min\n'b'\x1b[0;31mfmf/examples/wget/protocols/https\x1b[1;m\n\x1b[0;33mdescription\x1b[1;m: Test file download using the https protocol.\n\x1b[0;33mdetails\x1b[1;m: This is a very long and detailed description of what the wget/protocols/https test actually does including multiple examples and information about proper environment setup.\n\x1b[0;33mtags\x1b[1;m: Tier1\n\x1b[0;33mtest\x1b[1;m: runtest.sh\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n\x1b[0;33mtime\x1b[1;m: 1 min\n'b'\x1b[0;31mfmf/examples/wget/recursion/deep\x1b[1;m\n\x1b[0;33mdepth\x1b[1;m: 1000\n\x1b[0;33mdescription\x1b[1;m: Check recursive download options\n\x1b[0;33mtags\x1b[1;m: Tier2\n\x1b[0;33mtest\x1b[1;m: runtest.sh\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n\x1b[0;33mtime\x1b[1;m: 10 min\n'b'\x1b[0;31mfmf/examples/wget/recursion/fast\x1b[1;m\n\x1b[0;33mdepth\x1b[1;m: 10\n\x1b[0;33mdescription\x1b[1;m: Check recursive download options\n\x1b[0;33mtags\x1b[1;m: Tier2\n\x1b[0;33mtest\x1b[1;m: runtest.sh\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n\x1b[0;33mtime\x1b[1;m: 1 min\n'b'\x1b[0;31mfmf/examples/wget/requirements/download/output-document-file\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/download\n\x1b[0;33mpriority\x1b[1;m: medium\n\x1b[0;33mrequirement\x1b[1;m: Save content to a file.\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/requirements/download/output-document-pipe\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/download\n\x1b[0;33mpriority\x1b[1;m: medium\n\x1b[0;33mrequirement\x1b[1;m: Save content to pipe.\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/requirements/protocols/ftp\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/protocols/ftp\n\x1b[0;33mpriority\x1b[1;m: high\n\x1b[0;33mrequirement\x1b[1;m: Download a file using the ftp protocol.\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/requirements/protocols/http\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/protocols/http\n\x1b[0;33mpriority\x1b[1;m: high\n\x1b[0;33mrequirement\x1b[1;m: Download a file using the http protocol.\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/requirements/protocols/https\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/protocols/https\n\x1b[0;33mpriority\x1b[1;m: high\n\x1b[0;33mrequirement\x1b[1;m: Download a file using the https protocol.\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/requirements/upload/post-data\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/protocols/http\n\x1b[0;33mpriority\x1b[1;m: medium\n\x1b[0;33mrequirement\x1b[1;m: Upload a string to the server\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'b'\x1b[0;31mfmf/examples/wget/requirements/upload/post-file\x1b[1;m\n\x1b[0;33mcoverage\x1b[1;m: wget/protocols/http\n\x1b[0;33mpriority\x1b[1;m: medium\n\x1b[0;33mrequirement\x1b[1;m: Upload a file to the server\n\x1b[0;33mtester\x1b[1;m: Petr \xc5\xa0pl\xc3\xadchal <psplicha@redhat.com>\n'
```

There are a number of things to take into account.  Most importantly, the fact that `str.encode()` will always give you bytes on Python 3.

Reply:
Oh, I see. Thanks for pointing this out, Cleber. For now I fixed this in fa946fa. I tested all supported python versions. Should be working now fine.